Add Teleop to the Robot

a2aaron opened the pull request:
Blocked by PR #12 (Add Gamepad to the Robot)

a2aaron committed:
Add Gamepad to the Robot

a2aaron committed:
Add Drivetrain to the Robot

Drivetrain is LeftRightMotors renamed to a more descriptive name.

a2aaron committed:
Add TeleopCommand to the Robot

saltyskittlez commented on src/org/usfirst/frc/team1076/robot/Robot.java:
```diff
@@ -39,10 +47,11 @@
-		// chooser.addObject("My Auto", new MyAutoCommand());
-		SmartDashboard.putData("Auto mode", chooser);
+		gamepad = new Gamepad(0);
+        SmartDashboard.putNumber("Left Factor", 1);
```
Use the Smarter Dashboard; put default.

a2aaron replied:
This can be done when we merge the Smarter Dashboard PR

saltyskittlez commented on src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    public boolean execute() {    	
```
This method should be marked `@Override`

saltyskittlez commented on src/org/usfirst/frc/team1076/robot/commands/TeleopCommand.java:
```diff
@@ -0,0 +1,60 @@
+    	return isFinished();
+    }
+    
+    private double selectMaxAbs(double[] items) {
```
Use the `double...` type.

saltyskittlez commented on src/org/usfirst/frc/team1076/robot/subsystems/Drivetrain.java:
```diff
@@ -0,0 +1,66 @@
+		setSpeed(0);
+	}
+
+    public void initDefaultCommand() {
```
Do we need this code anymore?

saltyskittlez commented on testsrc/org/usfirst/frc/team1076/test/TestTeleopCommand.java:
```diff
@@ -0,0 +1,68 @@
+	public void testRotate() {
+		gamepad.lx = 1;
+		teleop.execute();
+		assertEquals("Left motor should be negative when turning left",
```
Which direction is the motor moving in?

saltyskittlez reviewed (changes requested):
`Put default` is a critical change.

a2aaron committed:
Minor cleanup in TeleopCommand and Drivetrain

- Marked `execute()` with `@Override`
- Changed `selectMaxAbs()` to use `double...` and not a double array
- Removed unused `initDefaultCommand()`

a2aaron committed:
Test rotation in both directions for TeleopCommand.

Additionally, clarify assertion messages.

a2aaron committed:
Merge branch 'master' into transfer/teleop

a2aaron committed:
Merge branch 'master' into transfer/teleop

a2aaron committed:
Use SmarterDashboard and not SmartDashboard

a2aaron committed:
Merge branch 'master' into transfer/teleop

saltyskittlez reviewed (approved):
👍
